quill-sql/src/test/scala/io/getquill/context/sql/idiom/SqlIdiomSpec.scala

```diff
@@ -844,6 +844,13 @@ class SqlIdiomSpec extends Spec {
     }
     "action" - {
       "insert" - {
+        "not affected by variable name" in {
+          val q = quote { (v: TestEntity) =>
+            query[TestEntity].insert(v)
+          }
+          val v = TestEntity("s", 1, 2L, Some(1))
+          testContext.run(q(lift(v))).string mustEqual "INSERT INTO TestEntity (s,i,l,o) VALUES (?, ?, ?, ?)"
```

Comment:
Should confirm whether this solution works with `Returning` and `Foreach`.

Reply:
Good catch! Actually when I introduced tests for `returning` and `foreach` I had to make some additional changes in `Parsing`!